src/context/user/user-handler.js

```diff
@@ -79,26 +84,31 @@ export function useInitializeUser() {
         );
         const paymentStatus = pageURL.searchParams.get("redirect_status");
 
+        // If payment succeeded, clear the cart in localStorage
+        if (paymentStatus === "succeeded") removeUserDataFromLocalStorage();
+
+        /**
+         * Combine the cart in localStorage with cart in database
+         * For an item that is in either of the carts (or both carts),
+         * the combined cart will use the larger number out of two carts
+         *
+         * This combined cart should replace both old carts in localStorage and database
+         */
         let userData;
         if (userSnap.exists()) {
           // user already exist, i.e user didn't sign up for the first time
           userData = userSnap.data();
           const cartInDb = userData.cart;
+          const mergedCart = mergeDbLocalCarts(cartInDb, localCart);
```

Comment:
There will need to be another check for whether the cart was cleared in the database. After the user successfully completed the payment, although we did update the cart data in database in the background, there could be a possibility that when we fetch data here the database hasn't completed the update yet, resulting in stale cart data.